src/worker/mod.rs

```diff
@@ -225,11 +228,23 @@ impl<E: StdError + 'static + Send> Worker<E> {
             .get(job.kind())
             .ok_or(Failed::BadJobType(job.kind().to_string()))?;
         match handler {
-            Callback::Async(cb) => cb.run(job).await.map_err(Failed::Application),
+            Callback::Async(cb) => AssertUnwindSafe(cb.run(job))
```

Comment:
Do we know that this is unwind safe? I feel like this should perhaps be a bound on what `cb.run` returns instead? Alternatively, can we `tokio::spawn` here, which will catch the panic for us?

Reply:
We do not and yes this should be a bound if we want to be safe here. I like your suggestion with `tokio::spawn` better. I remember this [comment](https://github.com/jonhoo/faktory-rs/pull/49#discussion_r1468867348) and so I think now _may_ want to spawn just to immediately await, we just should be unwrap the result, rather handle it properly.

The spawning here also brings me now back to this [thread](https://github.com/jonhoo/faktory-rs/pull/65#discussion_r1614435096) :smile:

Reply:
indeed I ended up using your earlier suggestion to avoid double-arc'ing